addons/account_peppol/models/account_edi_proxy_user.py

```diff
@@ -70,16 +70,17 @@ def _check_company_on_peppol(self, company, edi_identification):
         if (
             not company.account_peppol_migration_key
             and (participant_info := company.partner_id._get_participant_info(edi_identification)) is not None
-            and company.partner_id._check_peppol_participant_exists(participant_info, edi_identification, check_company=True)
+            and (provider := company.partner_id._check_peppol_participant_exists(participant_info, edi_identification, get_provider=True))
         ):
             error_msg = _(
                 "A participant with these details has already been registered on the network. "
                 "If you have previously registered to an alternative Peppol service, please deregister from that service, "
                 "or request a migration key before trying again. "
             )
 
-            if isinstance(participant_info, str):
-                error_msg += _("The Peppol service that is used is likely to be %s.", participant_info)
+            if isinstance(provider, str) and "Odoo" not in provider:
+                company.peppol_external_provider = provider
+                error_msg += _("The Peppol service that is used is likely to be %s.", provider)
```

Comment:
That part doesn't make any sense... You can not write something just before raising because it will rollback your write

Reply:
It works it you try...catch it. It's a bad idea to have that raise anyway in the first place... I'll try to clean that up too :face_with_spiral_eyes: